c2/c2-client-bundle/c2-client-api/src/main/java/org/apache/nifi/c2/client/api/C2Client.java

```diff
@@ -16,21 +16,20 @@
  */
 package org.apache.nifi.c2.client.api;
 
+import java.util.Optional;
 import org.apache.nifi.c2.protocol.api.C2Heartbeat;
 import org.apache.nifi.c2.protocol.api.C2HeartbeatResponse;
-import org.apache.nifi.c2.protocol.api.C2Operation;
 import org.apache.nifi.c2.protocol.api.C2OperationAck;
 
 import java.io.IOException;
 import java.nio.ByteBuffer;
-import java.util.List;
 
 /**
  * Defines interface methods used to implement a C2 Client. The controller can be application-specific but is used for such tasks as updating the flow.
  */
 public interface C2Client {
 
-    C2HeartbeatResponse publishHeartbeat(C2Heartbeat heartbeat) throws IOException;
+    Optional<C2HeartbeatResponse> publishHeartbeat(C2Heartbeat heartbeat) throws IOException;
```

Comment:
Looks like there's a checkstyle violation somewhere in this module, could've been there from me but just pointing it out as we'd like to be able to run with `-Pcontrib-check` before too long.

Reply:
Yeah it looks like there were a few inherited checkstyle violations I addressed all I could find (mostly unused import for C2Client references I suppose a remainder of the previous approach. contrib-check passes now.